virtcontainers/sandbox.go

```
@@ -574,8 +572,20 @@ func newSandbox(ctx context.Context, sandboxConfig SandboxConfig, factory Factor
 		}
 	}()
 
-	if err = s.hypervisor.createSandbox(ctx, s.id, s.networkNS, &sandboxConfig.HypervisorConfig, s.store); err != nil {
-		return nil, err
+	if s.supportNewStore() {
+		s.devManager = deviceManager.NewDeviceManager(sandboxConfig.HypervisorConfig.BlockDeviceDriver, nil)
+
+		// Ignore the error. Restore can fail for a new sandbox
+		s.Restore()
```

Why is it safe to ignore errors here? The old store relies on restore to determine if it should really create new guest. What is the workflow of newstore here?

---

They're almost same. `s.Restore()` always fail when it's first sandbox.

https://github.com/kata-containers/runtime/pull/1889/files#diff-bca2c2b5b04e0e74533b492fd23a3cc8R478 

this part will check if `s.state.State==""`, if state is empty, then we need to create a new sandbox; else this is an existing sandbox.

---

How about capturing the error and adding it to a `logger.WithError(err).Debug("found expected restore error")`?

---

@jodh-intel 
Actually I think it can be expected or unexpected, we can't know so we can't print debug message saying "it's expected"

---

Ack ;)